drivers/media/i2c/isx021.c

```
@@ -0,0 +1,1139 @@
+
+#include <media/v4l2-ctrls.h>
+#include <media/v4l2-fwnode.h>
+#include <media/v4l2-subdev.h>
+
+#define ISX021_WIDTH 1920
+#define ISX021_HEIGHT 1280
+
+#define PLUS_10(x) ((x) + (x) / 10)
+
+#define V4L2_CID_FSYNC (V4L2_CID_USER_BASE | 0x1002)
+#define V4L2_CID_TPG (V4L2_CID_USER_BASE | 0x1003)
+
+#define ISX021_SHUTTER_TIME_MIN 0
+#define ISX021_SHUTTER_TIME_MID 11010
+#define ISX021_SHUTTER_TIME_MAX 33000
+
+static const char *const isx021_supply_names[] = {
+	"dvdd",
+};
+
+static const char *const isx021_ctrl_fsync_options[] = {
+	"Internal",
+	"External",
+};
+
+static const char *const isx021_ctrl_test_pattern_options[] = {
+	"Disabled",
+	"Enabled",
+};
+
+struct isx021 {
+	struct device *dev;
+	struct clk *clk;
+	struct regulator_bulk_data supplies[ARRAY_SIZE(isx021_supply_names)];
+	struct gpio_desc *reset;
+	struct regmap *regmap;
+
+	bool streaming;
+	int trigger_mode;
+
+	struct v4l2_subdev subdev;
+	struct media_pad pad;
+
+	struct v4l2_ctrl_handler ctrls;
+};
+
+static int isx021_set_response_mode(struct isx021 *sensor);
+
+static inline struct isx021 *to_isx021(struct v4l2_subdev *sd)
+{
+	return container_of(sd, struct isx021, subdev);
+}
+
+static int isx021_read(struct isx021 *sensor, u16 addr, u8 *value)
+{
+	u32 reg_val = 0;
+	int ret;
+
+	ret = regmap_read(sensor->regmap, addr, &reg_val);
+	*value = reg_val & 0xFF;
+	if (ret)
+		dev_err(sensor->dev, "[%s] : Failed at 0x%x\n", __func__, addr);
+
+	return ret;
+}
+
+static int isx021_write(struct isx021 *sensor, u32 addr, u32 value)
+{
+	int ret = 0;
+
+	ret = regmap_write(sensor->regmap, addr, value);
+	if (ret)
+		dev_err(sensor->dev, "[%s] : Failed at 0x%x\n", __func__, addr);
+
+	return ret;
+}
+
+static int isx021_copy_reg_value(struct isx021 *sensor, u16 ae_addr,
+				 u16 me_addr)
+{
+	int ret = 0;
+	u8 val8 = 0;
+
+	ret = isx021_read(sensor, ae_addr, &val8);
+	if (ret)
+		dev_err(sensor->dev, "[%s] : Failed isx021_read_reg at 0x%x\n",
+			__func__, ae_addr);
```

Any special reason to ignore the error?

---

no, if error we should return